snap/snapcraft.yaml

```diff
@@ -0,0 +1,267 @@
+      - -bin/python3*
+
+  subiquitydeps:
+    plugin: nil
+    build-attributes: [enable-patchelf]
+    stage-packages:
+      # This list includes the dependencies for curtin and probert as well,
+      # there doesn't seem to be any real benefit to listing them separately.
+      - cloud-init
+      - dctrl-tools
+      - iso-codes
+      - libpython3-stdlib
+      - libpython3.10-minimal
+      - libpython3.10-stdlib
+      - libsystemd0
+      - lsb-release
+      - ntfs-3g
+      - python3-aiohttp
+      - python3-apport
+      - python3-attr
+      - python3-bson
+      - python3-debian
+      - python3-jsonschema
+      - python3-minimal
+      - python3-oauthlib
+      - python3-pkg-resources
+      - python3-pyroute2
+      - python3-pyrsistent
+      - python3-pyudev
+      - python3-requests
+      - python3-requests-unixsocket
+      - python3-systemd
+      - python3-urwid
+      - python3-yaml
+      - python3-yarl
+      - python3.10-minimal
+      - ssh-import-id
+      - ubuntu-advantage-tools
+      # WSL specifics:
+      - language-selector-common
+      - locales
+    prime:
+      - -lib/systemd/system/*
+
+  flutter-git:
+    source: .
+    override-pull: |
+      craftctl default
+      FLUTTER_VERSION=$(sed -n "s/^flutter \([0-9.]\+\).*/\1/p" .tool-versions)
+      git clone -b $FLUTTER_VERSION --depth 1 https://github.com/flutter/flutter.git
+    plugin: nil
+    override-build: |
+      mkdir -p $CRAFT_PART_INSTALL/usr/bin
+      mkdir -p $CRAFT_PART_INSTALL/usr/libexec
+      cp -r $CRAFT_PART_SRC/flutter $CRAFT_PART_INSTALL/usr/libexec/flutter
+      ln -s $CRAFT_PART_INSTALL/usr/libexec/flutter/bin/flutter $CRAFT_PART_INSTALL/usr/bin/flutter
+      ln -s $CRAFT_PART_INSTALL/usr/libexec/flutter/bin/dart $CRAFT_PART_INSTALL/usr/bin/dart
+      $CRAFT_PART_INSTALL/usr/bin/flutter doctor
+    build-packages:
+      - clang
+      - cmake
+      - curl
+      - libgtk-3-dev
+      - ninja-build
+      - unzip
+      - xz-utils
+      - zip
+    override-prime: ""
+
+  ubuntu-bootstrap:
+    after: [flutter-git]
+    source: .
+    source-type: git
+    plugin: nil
+    build-attributes: [enable-patchelf]
+    override-build: |
+      set -eux
+      mkdir -p $CRAFT_PART_INSTALL/bin/lib
+      cp snap/local/launcher $CRAFT_PART_INSTALL/bin/
+      cp snap/local/subiquity-server $CRAFT_PART_INSTALL/bin/
+      cp -r packages/subiquity_client/subiquity $CRAFT_PART_INSTALL/bin/
+      mkdir -p $CRAFT_PART_INSTALL/etc/subiquity
+      cp -r snap/local/postinst.d $CRAFT_PART_INSTALL/etc/subiquity
+      mkdir -p $CRAFT_PART_INSTALL/usr/share/applications
+      cp snap/local/ubuntu-desktop-installer.desktop $CRAFT_PART_INSTALL/usr/share/applications/
+      dart pub global activate melos
+      dart pub global run melos bootstrap
+      # https://github.com/canonical/ubuntu-desktop-installer/issues/1146
+      (cd packages/ubuntu_wizard && flutter pub get)
+      cd packages/ubuntu_bootstrap
+      flutter pub get
```

Comment:
The `pub get`s here shouldn't be needed, `melos bootstrap` has already taken care of those.